.github/workflows/basic_tests.yml

```diff
@@ -1,27 +1,29 @@
 # Runs all tests not flagged as "extended" with a pytest marker
-name: Tests
+name: Basic Tests
 
 defaults:
   run:
     shell: bash
 
-on:  [push]  # Runs on all push events to any branch
-
+on:  # Runs on all push events to any branch that isn't master
+  push:
+    branches-ignore:
+      - 'master'
 
 jobs:
   tests:
     name: ${{ matrix.os }} / ${{ matrix.python-version }}
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04, macos-latest, windows-latest]
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        os: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
This removes 2 ubuntu versions and 1 Python version. Is this intentional?

Reply:
yes. I put all of that into extended tests

Reply:
The basic tests should also test all python version and OSes

Reply:
Why? It's being tested before merging with all python versions and OSes anyway. 
I see it more like: 
Basic Tests: Did I break something obvious?
Extended Tests: Did I break something more advanced?